src/DbTarget.php

```diff
@@ -50,10 +49,9 @@ class DbTarget extends Target
      * This method will initialize the [[db]] property to make sure it refers to a valid DB connection.
      * @throws InvalidConfigException if [[db]] is invalid.
      */
-    public function init()
+    public function __construct()
     {
-        parent::init();
-        $this->db = Instance::ensure($this->db, Connection::class);
+        $this->db = Yii::ensureObject($this->db, Connection::class);
```

Comment:
When is $this->db set ? (The first arg to ensureObject)

Reply:
In this case it is suitable to pass `$db` to constructor with DI:

```php
public function __construct(ConnectionInterface $db)
{
    $this->db = $db;
}
```

Reply:
Done